server/src/main/java/com/broadcom/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -197,8 +202,14 @@ public void didChange(DidChangeTextDocumentParams params) {
   public void didClose(DidCloseTextDocumentParams params) {
     String uri = params.getTextDocument().getUri();
     LOG.info(format("Document closing invoked on URI %s", uri));
+    interruptAnalysis(uri);
     communications.publishDiagnostics(Map.of(uri, List.of()));
     docs.remove(uri);
+    clearAnalysedFutureObject(uri);
```

Comment:
removed communications.cancelProgressNotification and handled inside aspect. But from user perspective, we still see message on UI (analysis in progress) as we usually take more time than 3 sec to stop processing.

Reply:
We cannot control this message when it is already sent. It is expected behavior.